src/routes/ConsortiumManager/settings/authorization-roles/AuthorizationRolesViewPage.js

```diff
@@ -67,6 +77,28 @@ export const AuthorizationRolesViewPage = ({ path }) => {
   const userIds = useMemo(() => roles.map(i => i.metadata?.updatedByUserId), [roles]);
   const { users } = useUsers(userIds);
 
+  const onDuplicate = () => {
+    const role = roles.find(i => i.id === roleId);
```

Comment:
Please use `useRoleById` hook to get a role by id from `stripes-authorization-components` instead. If there is some pagination on the page, this behavior will be broken since roles might not include an ID.

Reply:
Moreover, it uses `react-query` that provides cached value